qiskit/transpiler/passes/routing/commuting_2q_gate_routing/commuting_2q_gate_router.py

```diff
@@ -164,6 +164,10 @@ def run(self, dag: DAGCircuit) -> DAGCircuit:
 
         current_layout = Layout.generate_trivial_layout(*dag.qregs.values())
 
+        self.property_set["layout"] = current_layout.copy()
+        self.property_set["original_qubit_indices"] = {
+            bit: index for index, bit in enumerate(dag.qubits)
+        }
```

Comment:
This pass seems to rely on the input circuit being physical already, which would imply that `self.property_set["layout"]` likely is already set, and this then overwrites it.

Is there a reason that this routing pass would want to be overwriting the layout and original qubit indices?

Reply:
> in practice we mostly use the pass in isolation

This is kind of where the problem is - if you want to use it in isolation, it should be on a circuit that was already physical, so either you need to have pretended that it was (in which case you should already know how to handle the missing layout), or the pass _should_ be running in a pass manager that makes the circuit physical via a `embed` pipeline.  Fwiw, I think `TranspileLayout` should handle the case of the missing `layout` fine already when using `routing_permutation`, which setting `final_layout` should automatically make available.

If we make this unilaterally set `layout` (to the trivial layout, no less) it becomes impossible to use correctly in a full pipeline, which I think might be a larger barrier to entry for general use.